### DISCUSSION TITLE
How to create an index through gremlin-client?

### DISCUSSION THREAD

**Discussion by zhengyingying** [JanusGraph/janusgraph#2652] [Q&A]
I try to create an index using gremlin-client to submit the following string. 

`graph = g.getGraph(); mgmt = graph.openManagement();mgmt.buildIndex('index-vl1-vp0', Vertex.class).addKey(mgmt.getPropertyKey('vp0')).buildCompositeIndex();mgmt.commit(); mgmt.awaitGraphIndexStatus(graph, 'index-vl1-vp0').call();
`

An Exception was thrown:
`Error during serialization: Class is not registered: org.janusgraph.graphdb.database.management.GraphIndexStatusReport
Note: To register this class use: kryo.register(org.janusgraph.graphdb.database.management.GraphIndexStatusReport.class);`

How can I solve it?

**Comment by vtslab** [comment]
I guess you used the conf/remot-object.yaml for connecting with the gremlin client, which specifies the JanusGraphIoRegistry for object serialization. Apparently, the object you want to return is not part of this registry.
You can either append a toString() to your statement or use the conf/remote.yaml for connecting with gremlin client.

**Reply by zhengyingying** [reply]
Thanks, I have solved it with conf/remote.yaml.

Another question: can I use `serializeResultToString: true` and `ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry]` at the same time in a yaml file?

**Reply by vtslab** [reply]
Specifying both in the yaml file is not possible, because there is no way to say per request which one to use. With remote-object.yaml you have the choice to do an explicit toString() for a request, either remotely (then you have only the string returned) or locally (then you have both the object and the string).
